### PR TITLE
fix: getSpan returning undefined, enable strict null checks

### LINKED ISSUE #360: TraceService.getSpan should return Span|undefined
This function

https://github.com/pragmaticivan/nestjs-otel/blob/da59285d80ee7b2ec6d4ea3941119d585a290d7c/src/tracing/trace.service.ts#L10

calls [this function](https://github.com/open-telemetry/opentelemetry-js/blob/main/api/src/trace/context-utils.ts#L34) in @opentelemetry/api which returns `Span | undefined`.

I'll submit a quick PR as well to just add the `| undefined`, we literally just ran into this in prod 😅

### PR DESCRIPTION
**Pull request by killercup**
Fixes #360

This changes the public API and is thus a breaking change.

Also enables strict null checks for the whole TypeScript project, without changing the API.

### CODE REVIEW

**Comment by killercup** [comment]
Not sure about your TSConfig but strict null checks should be able to catch this.

**Review by pragmaticivan — APPROVED** [review]
Thx!

**Comment by pragmaticivan** [comment]
@killercup could you explain why that is a breaking change? I only see default values changing, nothing in the contract.

**Comment by killercup** [comment]
Thanks for the quick merge!

I'm not sure what the right metric for breaking API change in TypeScript is tbh, if there is any. Assuming the user has strict null checks enabled, the following code will not compile anymore:

```ts
const span = this.traceService.getSpan();
span.addEvent("foo");
```

TypeScript now rightfully complains that `span` might be null. Instead, they either have to go with `span?.addEvent("foo");` or `if (span )span.addEvent("foo");` to make it compile.

So, long story short, updating might make your project stop compiling. This is what I'd call a breaking change :)
